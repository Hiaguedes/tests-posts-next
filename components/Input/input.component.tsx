import React, { useState } from 'react';
import { InputBase } from './input.styles';

interface InputProps {
    value: string;
    setValue: any;
    placeholder?: string;
}

const Input = ({value, setValue, placeholder}: InputProps) => {
    return (
        <div>
            <InputBase placeholder={placeholder} value={value} onChange={setValue}/>
        </div>
    );
}

export default Input;
