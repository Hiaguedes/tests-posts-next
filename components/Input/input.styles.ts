import styled from 'styled-components';

export const InputBase = styled.input`
    width: 70vw;
    border-radius: 1rem;
    height: 3rem;
    background-color: #f1f1f1;
    outline: none;
    padding: 0 2rem; 
    border: 1px solid rgba(0,0,0,0.4);
    color: rgb(0,0,0);

    ::placeholder {
        color: rgba(0,0,0,0.6);
    }
`;