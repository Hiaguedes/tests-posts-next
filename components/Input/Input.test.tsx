import Input from './input.component';
import { screen, render } from '@testing-library/react';

describe('Teste de snapshot do input', () => {

    test('Snapshot de 20/01/2021', () => {
        const {container} = render(<Input placeholder="Input de Test"/>)
        expect(container.firstChild).toMatchSnapshot();
    })
})