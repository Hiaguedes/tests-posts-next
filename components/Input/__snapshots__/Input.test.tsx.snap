// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot do input Snapshot de 20/01/2021 1`] = `
<div>
  <input
    class="inputstyles__InputBase-ctnzf7-0 fPuFyf"
    placeholder="Input de Test"
    value=""
  />
</div>
`;
